server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -1992,6 +2013,7 @@ public boolean flush(boolean force, boolean waitIfOngoing) throws EngineExceptio
                         // we need to refresh in order to clear older version values
                         refresh("version_table_flush", SearcherScope.INTERNAL, true);
                         translog.trimUnreferencedReaders();
+                        lastFlushTimestamp = relativeTimeInNanosSupplier.getAsLong();
```

Comment:
It feels safer to me to do this prior to the commit. That ensures that we get the next flush after 5 mins and not after 5 mins plus flush time and pause time to get here. In reality, it probably does not matter, so feel free to ignore if you think it is better here.